### ISSUE TITLE
[FA4, CuTe] ex2_emu breaks decode/prefill bitwise consistency on SM100 (MLA 192,128)

### ISSUE THREAD

**Issue by Luosuu** [Dao-AILab/flash-attention#2596]
`_TUNING_CONFIG` in b7 release enables ex2_emu_freq=32 for the causal MLA shape (use_2cta=False, is_causal=True, hdim=192, is_sm103=False) on SM100. This breaks bitwise consistency between decode-style attention (seqlen_q=1, paged KV) and prefill-style attention (seqlen_q=full, non-paged) at seqlen_k ≥ 512.

The polynomial ex2_emulation_2 activates based on a compile-time fragment counter in apply_exp2_convert. Different kernel variants (decode vs prefill) have different fragment counts, so the emulation activates in one but not the other. In b5, enable_ex2_emu was False for this shape — all exp2 used hardware exp2.approx and consistency was preserved.

## reproducer (sm100 only)

```python
import torch
from flash_attn.cute import flash_attn_varlen_func

seqlen_k, d, dv, nheads = 512, 192, 128, 16
device, dtype = "cuda", torch.bfloat16

torch.manual_seed(42)
q = torch.randn(seqlen_k, nheads, d, device=device, dtype=dtype)
k = torch.randn(seqlen_k, nheads, d, device=device, dtype=dtype)
v = torch.randn(seqlen_k, nheads, dv, device=device, dtype=dtype)

# Prefill
cu = torch.tensor([0, seqlen_k], dtype=torch.int32, device=device)
out_prefill, _ = flash_attn_varlen_func(
    q, k, v, cu_seqlens_q=cu, cu_seqlens_k=cu,
    max_seqlen_q=seqlen_k, max_seqlen_k=seqlen_k, causal=True)

# Decode (last position, paged KV)
page_size = 128
num_pages = (seqlen_k + page_size - 1) // page_size
k_c = torch.zeros(num_pages, page_size, nheads, d, device=device, dtype=dtype)
v_c = torch.zeros(num_pages, page_size, nheads, dv, device=device, dtype=dtype)
for i in range(seqlen_k):
    k_c[i // page_size, i % page_size] = k[i]
    v_c[i // page_size, i % page_size] = v[i]
pt = torch.arange(num_pages, dtype=torch.int32, device=device).unsqueeze(0)
out_decode, _ = flash_attn_varlen_func(
    q[-1:], k_c, v_c,
    cu_seqlens_q=torch.tensor([0, 1], dtype=torch.int32, device=device),
    cu_seqlens_k=None, max_seqlen_q=1, max_seqlen_k=None,
    seqused_k=torch.tensor([seqlen_k], dtype=torch.int32, device=device),
    page_table=pt, causal=True)

print(torch.equal(out_prefill[-1], out_decode[0]))
# b5: True  |  b7+: False (0.73% elements differ, max_diff=0.00049)
```

In softmax.py:apply_exp2_convert, the emulation decision depends on frg_cnt (fragment count) which varies with kernel variant:
- Decode kernel (seqlen_q=1, paged): small frg_cnt → boundary conditions force all hardware exp2
- Prefill kernel (seqlen_q≥384, non-paged): larger frg_cnt → some fragments use polynomial exp2
In b5 release, this shape had enable_ex2_emu=False (because 192 <= 128 is false in the default logic). The b7 _TUNING_CONFIG overrode it to True.

I made a PR #2595 to make ex2_emu optional so that user can have an option available for bitwise alignment between prefill/decode
